### PR TITLE
Upgrade to Node 12

### PR DESCRIPTION
**Pull request by emcsween**
### Description

Node 10 is EOL on April 30th 2021. We're moving to Node 12, which is going to be in maintenance until April 30th 2022.

#### Related Issues / PRs

overleaf/issues#3517

### Review

I grepped through for `.sort()` to see if we had bad sort predicates.

#### Potential Impact

Chat and comments

#### Manual Testing Performed

- [x] Send some messages through chat
- [x] Comment on a document

### CODE REVIEW

File: buildscript.txt [hunk 1]
```diff
@@ -3,6 +3,6 @@ chat
 --docker-repos=gcr.io/overleaf-ops
 --env-add=
 --env-pass-through=
---node-version=10.23.1
+--node-version=12.21.0
 --public-repo=False
---script-version=3.4.0
+--script-version=3.7.0
```
Comment (natestemen) [inline buildscript.txt:8]: Why did we need to update this?

**Review by natestemen — APPROVED** [review]
LGTM!

File: buildscript.txt (hunk 1 above)
Comment (natestemen) [reply buildscript.txt:8]: Presumably you ran the buildscript as the method to upgrade the node version. Just want to confirm though.

Comment (emcsween) [reply buildscript.txt:8]: yes. that's what happened

**Pull request merged**